nodeup/pkg/model/containerd.go

```diff
@@ -216,16 +200,35 @@ func (b *ContainerdBuilder) buildContainerOSConfigurationDropIn(c *fi.ModelBuild
 			{"systemctl", "restart", "kops-configuration.service", "&"},
 		},
 	})
+}
 
-	if err := b.buildSysconfig(c); err != nil {
-		return err
+// buildSystemdServiceOverrideFlatcar is responsible for overriding the containerd service for Flatcar
+func (b *ContainerdBuilder) buildSystemdServiceOverrideFlatcar(c *fi.ModelBuilderContext) {
+	lines := []string{
+		"[Service]",
+		"Environment=CONTAINERD_CONFIG=/etc/containerd/config-kops.toml",
+		"EnvironmentFile=/etc/environment",
 	}
+	contents := strings.Join(lines, "\n")
 
-	return nil
+	c.AddTask(&nodetasks.File{
+		Path:     "/etc/systemd/system/containerd.service.d/10-kops.conf",
+		Contents: fi.NewStringResource(contents),
+		Type:     nodetasks.FileType_File,
+		OnChangeExecute: [][]string{
+			{"systemctl", "daemon-reload"},
+			{"systemctl", "restart", "containerd.service"},
+			// We need to restart kops-configuration service since nodeup needs to load images
+			// into containerd with the new config. Restart is on the background because
+			// kops-configuration is of type 'one-shot' so the restart command will wait for
+			// nodeup to finish executing
```

Comment:
```suggestion
			// into containerd with the new config. We restart in the background because
			// kops-configuration is of type "one-shot," so the restart command will wait for
			// nodeup to finish executing.
```

Reply:
Thanks. This is embarrassing. Just copy pasted this without reading. 😁